controllers/provider-aws/pkg/validator/shoot_handler.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"context"
+	"net/http"
+
+	provideraws "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/aws"
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// Shoot validates shoots
+type Shoot struct {
+	client  client.Client
+	decoder runtime.Decoder
+	Logger  logr.Logger
+}
+
+// Handle implements Handler.Handle
+func (v *Shoot) Handle(ctx context.Context, req admission.Request) admission.Response {
+	shoot := &gardencorev1alpha1.Shoot{}
+
+	_, _, err := v.decoder.Decode(req.Object.Raw, nil, shoot)
+	if err != nil {
+		v.Logger.Error(err, "failed to decode shoot", string(req.Object.Raw))
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if shoot.Spec.Provider.Type != provideraws.Type {
+		return admission.Allowed("webhook not responsible for this provider")
+	}
+
+	if req.Operation == admissionv1beta1.Update {
+		oldShoot := &gardencorev1alpha1.Shoot{}
+		_, _, err := v.decoder.Decode(req.OldObject.Raw, nil, oldShoot)
+		if err != nil {
+			v.Logger.Error(err, "failed to decode old shoot", string(req.OldObject.Raw))
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+
+		if err := v.validateShootUpdate(ctx, oldShoot, shoot); err != nil {
+			v.Logger.Error(err, "denied request")
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+	}
```

Comment:
```suggestion

	return admission.Allowed("validations succeeded")
	}
```

I guess, there should be a "successful" return statement here, as `validateShoot(ctx, shoot)` is already called at the end of the `validateShootUpdate` function. Am I right?